code/controllers/subsystems/ghostroles.dm

```diff
@@ -0,0 +1,176 @@
+			log_debug("Spawner [G.type] got removed from selection because of missing data")
+			continue
+		//Check if we have a spawnpoint on the current map
+		if(!G.select_spawnpoint(FALSE))
+			log_debug("Spawner [G.type] got removed from selection because of missing spawnpoint")
+			continue
+		LAZYSET(spawners, G.short_name, G)
+	
+	for (var/identifier in spawnpoints)
+		CHECK_TICK
+		update_spawnpoint_status_by_identifier(identifier)
+
+//Adds a spawnpoint to the spawnpoint list
+/datum/controller/subsystem/ghostroles/proc/add_spawnpoints(var/obj/effect/ghostspawpoint/P)
+	if(!P.identifier) //If the spawnpoint has no identifier -> Abort
+		log_ss("ghostroles","Spawner [P] at [P.x],[P.y],[P.z] has no identifier set")
+		qdel(P)
+		return
+
+	if(!spawnpoints[P.identifier])
+		spawnpoints[P.identifier] = list()
+	
+	spawnpoints[P.identifier].Add(P)
+	//Only update the status if the round is started. During initialization that´s taken care of at the end of init.
+	if(ROUND_IS_STARTED)
+		update_spawnpoint_status(P)
+
+/datum/controller/subsystem/ghostroles/proc/update_spawnpoint_status(var/obj/effect/ghostspawpoint/P)
+	if(!P || !istype(P))
+		return null
+	//If any of the spawners uses that spawnpoint and is active set the status to available
+	for(var/s in spawners)
+		var/datum/ghostspawner/G = spawners[s]
+		if(!G.is_enabled())
+			continue
+		if(!(P.identifier in G.spawnpoints))
+			continue
+		P.set_available()
+		return TRUE
+	P.set_unavailable()
+	return FALSE
+
+/datum/controller/subsystem/ghostroles/proc/update_spawnpoint_status_by_identifier(var/identifier)
+	if(!identifier) //If no identifier ist set, return false
+		return null
+	if(!length(spawnpoints[identifier])) //If we have no spawnpoints for that identifier, return false
+		return null
+	
+	for (var/spawnpoint in spawnpoints[identifier])
+		CHECK_TICK
+		var/obj/effect/ghostspawpoint/P = spawnpoint
+		update_spawnpoint_status(P)
+
+/datum/controller/subsystem/ghostroles/proc/remove_spawnpoints(var/obj/effect/ghostspawpoint/G)
+	spawnpoints[G.identifier].Remove(G)
+	return
+
+//Returns the turf where the spawnpoint is located and updates the spawner to be used
+/datum/controller/subsystem/ghostroles/proc/get_spawnpoint(var/identifier, var/use = TRUE)
+	if(!identifier) //If no identifier ist set, return false
+		return null
+	if(!length(spawnpoints[identifier])) //If we have no spawnpoints for that identifier, return false
+		return null
+
+	for (var/spawnpoint in spawnpoints[identifier])
+		var/obj/effect/ghostspawpoint/P = spawnpoint
+		if(P.is_available())
+			if(use)
+				P.set_spawned()
+			return get_turf(P)
+
+/datum/controller/subsystem/ghostroles/proc/vui_interact(mob/user,var/spawnpoint=null)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user,src)
+	if(!ui)
+		ui = new(user,src,"misc-ghostspawner",950,700,"Ghost Role Spawner", nstate = interactive_state)
+		ui.data = vueui_data_change(list("spawnpoint"=spawnpoint,"current_tag"="All"),user,ui)
```

Comment:
you now can actually use new thing I added - `ui,metadata` for this. It migth be nicier and better way to do spawnpoint lockdown.

Reply:
After discussion in discord, opted not to implement that.